nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -220,10 +220,9 @@ let
         ++ map escapeShellArg container.cmd
       );
 
-      ExecStartPre = ["-${pkgs.docker}/bin/docker rm -f ${name}"
-                      "-${pkgs.docker}/bin/docker image prune -f"] ++
```

Comment:
I wasn't able to find out why this was added in the first place https://github.com/NixOS/nixpkgs/commit/508343962ea26e9da9a4c0c10009bfba85f50021#diff-a632ec0d1be978f9f4d3736c79cc51c3R224.

I thought maybe as a mean to ensure the image is always pulled, however it couldn't have worked provided the image needs to be in a dangling state [(SO about dangling images)](https://stackoverflow.com/a/45143234/1806023).

A better way to ensure the latest image is used (to avoid outdated local cache) would be to always pull the image, which @mkaito added in the same PR, but then removed due to this [comment](https://github.com/NixOS/nixpkgs/pull/78366#discussion_r371124557) because it was systematically pulling. While we may not want to always pull the image, making this available as an option is a very valid use case.

Reply:
Hmm, good thought.  I had assumed it was done to avoid leaving stale images around when it gets updated but perhaps it had more to do with the image pulling policy.

Kubernetes pods have an `imagePullPolicy` field that you can set to `Always`, `IfNotPresent`, or `Never`.  We could emulate the first two of those easily, but `Never` is harder unless `docker run` has a flag that I haven't seen to do it.

I'll look into this a bit and open another PR to add an option to control pulling at some point.